ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/linear_optimisation/algorithms/fillers/MaxMinRelativeMarginFiller.java

```diff
@@ -43,32 +42,18 @@ public MaxMinRelativeMarginFiller(Set<FlowCnec> optimizedCnecs,
         this.initialFlowResult = initialFlowResult;
         this.unit = unit;
         this.ptdfSumLowerBound = maxMinRelativeMarginParameters.getPtdfSumLowerBound();
-        this.negativeMarginObjectiveCoefficient = maxMinRelativeMarginParameters.getNegativeMarginObjectiveCoefficient();
+        this.highestThreshold = maxMinRelativeMarginParameters.getHighestThresholdValue();
     }
 
     @Override
     public void fill(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult) {
         super.fill(linearProblem, flowResult, sensitivityResult);
-        updateMinimumNegativeMarginDefinitionAndCost(linearProblem);
         buildMinimumRelativeMarginVariable(linearProblem);
+        buildMinimumRelativeMarginSignBinaryVariable(linearProblem);
         buildMinimumRelativeMarginConstraints(linearProblem);
         fillObjectiveWithMinRelMargin(linearProblem);
     }
 
-    /**
-     * Force the minimum margin variable (absolute margin) to be negative (unsecured case)
-     * Add a big coefficient to it in the objective function, in order to render it the primary objective
-     */
-    private void updateMinimumNegativeMarginDefinitionAndCost(LinearProblem linearProblem) {
```

Comment:
I don't think you should have removed this, it allows forcing the MM variable defined in MaxMinMarginFiller to be negative. 
You must change it however to use the new binary variable, and skip changing its coefficient in the objective function
( (1 - binary) * minimum absolute RAM <= MM <= 0)

Reply:
instead of doing it in the same method as MRM, do it here for better readability